tardis/visualization/tools/sdec_plot.py

```diff
@@ -1178,6 +1183,28 @@ def generate_plot_mpl(
                 linewidth=1,
             )
 
+        # Plot observed spectrum
+        if observed_spectrum:
+            if distance is None:
+                raise ValueError(
+                    "Distance must be specified if an observed_spectrum is given"
```

Comment:
... so that the model spectrum can be converted into flux space correctly"

Might be helpful to have this error message say something more informative, such as what I have suggested here?